meta-balena-common/recipes-core/systemd/timeinit/timeinit-rtc.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/sh
+#
+# Copyright 2020 Balena Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+. /usr/libexec/os-helpers-logging
+. /usr/libexec/os-helpers-time
+
+RTC_DEVICE=/dev/rtc
+
+# Don't bother updating or reporting errors for small differences.
+TIME_DIFF_THRESHOLD=2
+
+info "Setting system time from RTC."
+
+if [ ! -e "$RTC_DEVICE" ]; then
+	fail "RTC device ${RTC_DEVICE} not found."
```

Comment:
I did originally have this as a 'warn', but the script can't actually be called by systemd if `/dev/rtc` doesn't exist...so this is just a trap in case the user tries to run the script directly from the command line. I thought a 'fail' was best in this case as the script is unable to continue.